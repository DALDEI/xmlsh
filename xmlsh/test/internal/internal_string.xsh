# 
# Test internal boolean functions
. ../common

message  string() should be empty not null
set string()
[ $# -eq 1 ] && message  Pass 1 arg || message  fail $# args
shift
[ -z string() ] && message  Pass true || message  fail true
error 'string() might need to evaluate to nothing like $s'

message string(Hi There) should be Hi There
[ string(Hi There) = "Hi There" ] && message  Pass true || message  fail true

s=string()
message  $s should be empty not null
## set $s
error 'set $s fails for now'
[ $# -eq 1 ] && message  Pass 1 arg || message  fail $# args
s=string(Hi There)

message $s should be Hi There
[ "$s" = "Hi There" ] && message  Pass true || message  fail true
