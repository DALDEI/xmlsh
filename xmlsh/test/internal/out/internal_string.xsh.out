[5]  should be empty not null
[7] Pass 1 arg
[9] Pass true
[12] Hi There should be Hi There
[13] Pass true
[16] should be empty not null
[19] fail 0 args
[22] Hi There should be Hi There
[23] Pass true
